### PR TITLE
[kombucha] Fix sudoku test

### PR DESCRIPTION
**Pull request by katat**
This PR aims to fix the sudoku test, so that the test can be used as a baseline for other changes.

- fix some of the kombucha APIs
- fix the sudoku test
- use a new sudoku example as the previous one seems to have no solution for diagonal rules
- the "equals" API seems to have WIP code, for now this PR reverts the WIP changes in the "equals" to previous code

### CODE REVIEW

File: circuit-construction/tests/sudoku.rs [hunk 1]
```diff
@@ -55,17 +42,17 @@ where
...
+            vec![e, e, 2, /* */ 4, e, e, /* */ 1, e, e],
+            vec![4, e, e, /* */ e, 2, e, /* */ e, e, 7],
+            vec![e, e, 7, /* */ e, e, 3, /* */ 8, e, e],
             /* -------------------------------------- */
-            vec![e, 6, e, /* */ e, e, e, /* */ 2, 8, e],
-            vec![e, e, e, /* */ 4, 1, 9, /* */ e, e, 5],
-            vec![e, e, e, /* */ e, 8, e, /* */ e, 7, 9],
+            vec![1, e, 8, /* */ 2, e, e, /* */ e, e, e],
+            vec![e, 7, e, /* */ 1, e, 8, /* */ 4, e, e],
+            vec![e, 2, e, /* */ e, e, 9, /* */ e, 1, e],
```
Comment (mimoo) [inline circuit-construction/tests/sudoku.rs:55]: why are you changing the sudoku :D ? it was not correct? (I took it from wikipedia without really verifying haha)

Comment (mimoo) [reply circuit-construction/tests/sudoku.rs:55]: > use a new sudoku example as the previous one seems to have no solution for diagonal rules

ah!

**Comment by mimoo** [comment]
wait does the sudoku work now? It was just the sub that was wrong?

**Comment by katat** [comment]
> wait does the sudoku work now? It was just the sub that was wrong?
The sudoku test passes now. 

There is a [fix](https://github.com/o1-labs/proof-systems/pull/634/commits/73e064400a84cba5e9d83d849fd57c2c74a858c6) to the "equals" API. 
Then there is some verification [fixes](https://github.com/o1-labs/proof-systems/pull/634/commits/da25e20bec1d04b86eb4e03c531a0e26a082cd21) in the sudoku test.

**Review by mimoo — APPROVED** [review]
\o/

**Pull request merged**